geode-redis/src/integrationTest/java/org/apache/geode/redis/StringsIntegrationTest.java

```diff
@@ -219,30 +351,313 @@ public void testMSetAndGet() {
+    Future<Integer> future1 = pool.submit(callable1);
+    Future<Integer> future2 = pool.submit(callable2);
+
+    latch.countDown();
+
+    future1.get();
+    future2.get();
+
+    assertThat(jedis.get("contestedKey")).isEqualTo(Integer.toString(-2 * ITERATION_COUNT));
+  }
+
+  private Integer doABunchOfDecrs(Jedis jedis, CountDownLatch latch) throws InterruptedException {
+    latch.await();
+    for (int i = 0; i < ITERATION_COUNT; i++) {
+      jedis.decr("contestedKey");
+    }
+    return ITERATION_COUNT;
+  }
+
+  @Test
+  public void testIncr() {
+    String oneHundredKey = randString();
+    String negativeOneHundredKey = randString();
+    String unsetKey = randString();
+    final int oneHundredValue = 100;
+    final int negativeOneHundredValue = -100;
+    jedis.set(oneHundredKey, Integer.toString(oneHundredValue));
+    jedis.set(negativeOneHundredKey, Integer.toString(negativeOneHundredValue));
+
+    jedis.incr(oneHundredKey);
+    jedis.incr(negativeOneHundredKey);
+    jedis.incr(unsetKey);
+
+    assertThat(jedis.get(oneHundredKey)).isEqualTo(Integer.toString(oneHundredValue + 1));
+    assertThat(jedis.get(negativeOneHundredKey))
+        .isEqualTo(Integer.toString(negativeOneHundredValue + 1));
+    assertThat(jedis.get(unsetKey)).isEqualTo(Integer.toString(1));
+  }
+
+  @Test
+  public void testIncr_whenOverflow_shouldReturnError() {
+    String key = "key";
+    String max64BitIntegerValue = "9223372036854775807";
+    jedis.set(key, max64BitIntegerValue);
+
+    try {
+      jedis.incr(key);
+    } catch (JedisDataException e) {
+      assertThat(e.getMessage()).contains("value is not an integer or out of range");
+    }
+    assertThat(jedis.get(key)).isEqualTo(max64BitIntegerValue);
+  }
+
+  @Test
+  public void testIncr_whenWrongType_shouldReturnError() {
+    String key = "key";
+    String nonIntegerValue = "I am not a number! I am a free man!";
+    jedis.set(key, nonIntegerValue);
+
+    try {
+      jedis.incr(key);
+    } catch (JedisDataException e) {
+      assertThat(e.getMessage()).contains(RedisConstants.ERROR_WRONG_TYPE);
+    }
+    assertThat(jedis.get(key)).isEqualTo(nonIntegerValue);
+  }
+
+  @Test
+  public void testIncr_shouldBeAtomic() throws ExecutionException, InterruptedException {
+    jedis.set("contestedKey", "0");
+
+    CountDownLatch latch = new CountDownLatch(1);
+    ExecutorService pool = Executors.newFixedThreadPool(2);
+    Callable<Integer> callable1 = () -> doABunchOfIncrs(jedis, latch);
+    Callable<Integer> callable2 = () -> doABunchOfIncrs(jedis2, latch);
+    Future<Integer> future1 = pool.submit(callable1);
+    Future<Integer> future2 = pool.submit(callable2);
+
+    latch.countDown();
+
+    future1.get();
+    future2.get();
 
-    assertTrue(response == 1);
+    assertThat(jedis.get("contestedKey")).isEqualTo(Integer.toString(2 * ITERATION_COUNT));
+  }
+
+  @Test
+  // @todo: not looked over for current release
```

Comment:
There are a couple of these comments. Since all of these good changes are already being applied to this class, is there any reason not to validate these remaining few tests? Or is this more work than what it appears on the surface?

Reply:
I think this story had gone on a bit longer than was originally planned and the scope kept expanding, so creating stories to go back and review the implementation/tests for the commands that weren't part of this release was a way of reining in the scope.  @ringles may be able to provide more context though.

Reply:
The issue is that there are higher-priority commands that need love and attention before we circle back to these. There are stories in the backlog for them, but they aren't our current top priority.